### PR TITLE
Replace p1_time_compat with direct OTP functions

### PR DESCRIPTION
**Pull request by NelsonVides**
Pretty much a mechanical change. Need to see results from CI.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7342.1](https://travis-ci.org/esl/MongooseIM/jobs/599157914) / Erlang 22.0 / small_tests / 6c7c4a2360b61d03becc486e2af460e048d06222
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2498/7342/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2498/7342/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-17_13.49.49/lib.mongooseim.logs/run.2019-10-17_13.49.50/suite.log.html)

**Comment by mongoose-im** [comment]
[7343.1](https://travis-ci.org/esl/MongooseIM/jobs/599173453) / Erlang 22.0 / small_tests / 05f0f102c69ed12629c5d0356dfa44b7dc8fbf92
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2498/7343/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2498/7343/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-17_14.11.46/lib.mongooseim.logs/run.2019-10-17_14.11.47/suite.log.html)

**Comment by mongoose-im** [comment]
[7344.1](https://travis-ci.org/esl/MongooseIM/jobs/599181070) / Erlang 22.0 / small_tests / 1347427734c0f263cf68e9f44cc21deda813f9cf
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2498/7344/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2498/7344/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-17_14.26.50/lib.mongooseim.logs/run.2019-10-17_14.26.51/suite.log.html)

File: src/global_distrib/mod_global_distrib_bounce.erl [hunk 1]
```diff
@@ -103,10 +103,10 @@ maybe_store_message({From, To, Acc0, Packet} = FPacket) ->
             ?DEBUG("Storing global message id=~s from=~s to=~s to "
                    "resend after ~B ms (bounce_ttl=~B)",
                    [ID, jid:to_binary(From), jid:to_binary(To),
-                    p1_time_compat:convert_time_unit(opt(resend_after), native, milli_seconds),
+                    erlang:convert_time_unit(opt(resend_after), native, milli_seconds),
```
Comment (michalwski) [inline src/global_distrib/mod_global_distrib_bounce.erl]: Deprecated `milli_seconds`

File: src/global_distrib/mod_global_distrib_worker.erl [hunk 2]
```diff
@@ -51,8 +51,8 @@ handle_cast({route, {From, To, Acc, Packet}}, State) ->
     ejabberd_router:route(From, To, Acc, Packet),
     {noreply, State, ?TIMEOUT};
 handle_cast({data, Host, TransferTime, Stamp, Data}, State) ->
-    QueueTimeNative = p1_time_compat:monotonic_time() - Stamp,
-    QueueTimeUS = p1_time_compat:convert_time_unit(QueueTimeNative, native, micro_seconds),
+    QueueTimeNative = erlang:monotonic_time() - Stamp,
+    QueueTimeUS = erlang:convert_time_unit(QueueTimeNative, native, micro_seconds),
```
Comment (michalwski) [inline src/global_distrib/mod_global_distrib_worker.erl]: Deprecated `micro_seconds`.

File: src/mam/mam_iq.erl [hunk 3]
```diff
@@ -207,7 +207,7 @@ form_to_lookup_params(#iq{sub_el = QueryEl} = IQ, MaxResultLimit, DefaultResultL
 common_lookup_params(QueryEl, MaxResultLimit, DefaultResultLimit) ->
     RSM = fix_rsm(jlib:rsm_decode(QueryEl)),
     Limit = elem_to_limit(QueryEl),
-    #{now => p1_time_compat:system_time(micro_seconds),
+    #{now => erlang:system_time(micro_seconds),
```
Comment (michalwski) [inline src/mam/mam_iq.erl]: Deprecated `micro_seconds`.

File: src/mod_last.erl [hunk 4]
```diff
@@ -131,7 +131,7 @@ process_local_iq(_From, _To, Acc,
 get_node_uptime() ->
     case ejabberd_config:get_local_option(node_start) of
         {_, _, _} = StartNow ->
-            now_to_seconds(p1_time_compat:timestamp()) - now_to_seconds(StartNow);
+            erlang:system_time(seconds) - now_to_seconds(StartNow);
```
Comment (michalwski) [inline src/mod_last.erl]: I realized just now, that `seconds` is also deprecated time unit :) the correct one is just `second` without the `s`. I'm not going back to all the previous places where `seconds` unit were used, I sure you'll find them easily. Also I stop pointing other deprecated time units unless I spot one I didn't comment yet.

**Comment by NelsonVides** [comment]
@fenek small tests done, sorry I forgot that :sweat_smile: 
@michalwski about user, I was having a look at that, and I think it gets quite complicated, I think out of the scope of `p1_time_compat`. It's used in tons of places and the values it returns are passed along a ton of others, not a mechanical change perhaps :disappointed: 
Also, the damn deprecated time_units, that surprised me xD

**Comment by mongoose-im** [comment]
[7348.1](https://travis-ci.org/esl/MongooseIM/jobs/599343872) / Erlang 22.0 / small_tests / 98d94a68ac75969416080d459fb9b74486c5b035
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2498/7348/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2498/7348/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-17_20.45.31/lib.mongooseim.logs/run.2019-10-17_20.45.32/suite.log.html)

File: src/ejabberd_s2s_out.erl [hunk 5]
```diff
@@ -1151,7 +1151,7 @@ wait_before_reconnect(StateData) ->
                 undefined_delay ->
                     %% The initial delay is random between 1 and 15 seconds
                     %% Return a random integer between 1000 and 15000
-                    {_, _, MicroSecs} = p1_time_compat:timestamp(),
+                    MicroSecs = erlang:system_time(microsecond),
```
[the remark this answers is not in the document]
Comment (NelsonVides) [reply src/ejabberd_s2s_out.erl]: I'm not sure this makes much of a difference, it's not a place were cryptographically safe randomization makes any difference. We just want to wait some random amount of time between 1 an 15 seconds. The fact that this function might run at unpredictable times makes it sufficiently random for our purposes I think :slightly_frowning_face: 
An option would be to do `X = rand:uniform(14000), X+1000`, but the performance is pretty much the same (I've just measure it on my laptop with `timer:tc` over a 10M repeats), and `rand` is saving seeds into the process dictionary. If you prefer that I can quickly change it, but I don't think it matters much :man_shrugging:

**Comment by mongoose-im** [comment]
[7349.1](https://travis-ci.org/esl/MongooseIM/jobs/599491617) / Erlang 22.0 / small_tests / f407a76eff15a2b38a6f72abedcdac34f98d1640
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2498/7349/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2498/7349/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-18_06.51.33/lib.mongooseim.logs/run.2019-10-18_06.51.33/suite.log.html)

**Pull request merged**